### PR TITLE
fix(state manager): Make viewport-related data optional for annotation event

### PR DESCRIPTION
**Pull request by igoroctaviano**
- The event was not being triggered by the FOR state manager
- The annotation event is being used by the measurement service [here](https://github.com/OHIF/Viewers/blob/eb7c950d7e0dee35c0c51ac77e549798278ddcc4/extensions/cornerstone/src/initMeasurementService.js#L149) and it's not required to have viewport-related data since it's being consumed without it. This change makes it optional.

- Related PR: https://github.com/OHIF/Viewers/pull/3857

#### PR

- [] My Pull Request title is descriptive, accurate and follows the
  semantic-release format and guidelines.

#### Code

- [] My code has been well-documented (function documentation, inline comments,
  etc.)

- [] I have run the `yarn build:update-api` to update the API documentation, and have
  committed the changes to this PR. (Read more here https://www.cornerstonejs.org/docs/contribute/update-api)

#### Public Documentation Updates

- [] The documentation page has been updated as necessary for any public API
  additions or removals.

#### Tested Environment

- [] "OS: 
- [] "Node version: 
- [] "Browser:
  

[blog]: https://circleci.com/blog/triggering-trusted-ci-jobs-on-untrusted-forks/
[script]: https://github.com/jklukas/git-push-fork-to-upstream-branch

### CODE REVIEW

**Comment by igoroctaviano** [comment]
@sedghi @wayfarer3130 can you please review this one?

File: packages/tools/src/stateManagement/annotation/FrameOfReferenceSpecificAnnotationManager.ts [hunk 1]
```diff
@@ -233,6 +234,8 @@ class FrameOfReferenceSpecificAnnotationManager implements IAnnotationManager {
     toolSpecificAnnotations.push(annotation);
     checkAndDefineIsLockedProperty(annotation);
     checkAndDefineIsVisibleProperty(annotation);
+
+    triggerAnnotationAddedForFOR(annotation);
```
Comment (sedghi) [inline packages/tools/src/stateManagement/annotation/FrameOfReferenceSpecificAnnotationManager.ts:238]: we don't trigger events inside here, so I prefer to do this inside the state.

File: packages/tools/src/types/EventTypes.ts [hunk 2]
```diff
@@ -104,9 +104,9 @@ type ToolActivatedEventDetail = {
  */
 type AnnotationAddedEventDetail = {
   /** unique id of the viewport */
-  viewportId: string;
+  viewportId?: string;
```
Comment (sedghi) [inline packages/tools/src/types/EventTypes.ts:107]: Yes, this should have been like this from the beginning, but we accidentally linked it to the Ids :()

File: packages/tools/src/stateManagement/annotation/helpers/state.ts [hunk 3]
```diff
@@ -63,19 +61,17 @@ function triggerAnnotationAddedForFOR(annotation: Annotation) {
     });
   });
 
+  const eventType = Events.ANNOTATION_ADDED;
+  const eventDetail: AnnotationAddedEventDetail = { annotation };
+
   if (!viewportsToRender.length) {
+    triggerEvent(eventTarget, eventType, eventDetail);
```
Comment (sedghi) [inline packages/tools/src/stateManagement/annotation/helpers/state.ts:68]: So by these changes are firing two events? why?

File: packages/tools/src/stateManagement/annotation/FrameOfReferenceSpecificAnnotationManager.ts [hunk 4]
```diff
@@ -5,6 +5,7 @@ import {
   AnnotationState,
   GroupSpecificAnnotations,
 } from '../../types/AnnotationTypes';
+import { triggerAnnotationAddedForFOR } from './helpers/state';
```
Comment (sedghi) [inline packages/tools/src/stateManagement/annotation/FrameOfReferenceSpecificAnnotationManager.ts:8]: remove please

**Review by sedghi — APPROVED** [review]

**Pull request merged**